R/featureMethods.R

```diff
@@ -0,0 +1,246 @@
+#' @param min_subunits Integer specifying minimum number of subunits in a co-elution feature.
+#' @param min_peak_corr Numeric value betwee 0 and 1 specifying minimum peak correlation, defaults to 0.5.
+#' @param min_monomer_distance_factor Numeric value specifying a factor to multiply the largest monomer molecular weight, defaults to \code{NULL}.
+#' This filters out features that have their apex at a smaller molecular weight than the resulting min_monomer_distance_factor*max(monomer_mw) value.
+#' Using this filtering option can, for example, remove complex features that are likely spontaneous co-elutions of the subunits monomers.
+#' @return The same feture table as teh input, but filtered according to the provided parameters.
+#' @examples
+#' ## Load example complex feature finding results:
+#' complexFeatures <- exampleComplexFeatures
+#' ## Run summary function:
+#' summarizeFeatures(complexFeatures)
+#' ## Filter complex features by a peak correlation of 0.5, a minimum hypothesis
+#' ## completeness of 0.5 and a minimum distance to the monomers by a factor of 2:
+#' filteredComplexFeatures <- filterFeatures(complexFeatures, 
+#'                                      min_peak_corr=0.5, 
+#'                                      min_hypothesis_completeness=0.5,
+#'                                      min_monomer_distance_factor=2)
+#' ## Run summary function on filtered data:
+#' summarizeFeatures(filteredComplexFeatures)
+#' @export
+filterFeatures <- function(feature_table,
+                           complex_ids=NULL,
+                           protein_ids=NULL,
+                           min_feature_completeness=NULL,
+                           min_hypothesis_completeness=NULL,
+                           min_subunits=NULL,
+                           min_peak_corr=0.5,
+                           min_monomer_distance_factor=NULL){
+  if("complex_id" %in% names(feature_table)){
+    type="complex"
+  } else if ("protein_id" %in% names(feature_table)) {
+    type="protein"
+  } else {
+    stop("This is neither a complex or a protein feature table. Please check the input data.")
+  }
+  if(!is.null(complex_ids)){
+    if(type=="complex"){
+      feature_table <- subset(feature_table,complex_id %in% as.character(complex_ids))
+    } else {
+      stop("This is not a complex feature table. Please check the input data and your filtering criteria, especially the complex_ids.")
+    }
+  }
+  if(!is.null(protein_ids)){
+    if(type=="protein"){
+      feature_table <- subset(feature_table,protein_id %in% as.character(protein_ids))
+    } else {
+      stop("This is not a protein feature table. Please check the input data and your filtering criteria, especially the protein_ids.")
+    }
+  }
+  if(!is.null(min_feature_completeness)){
+    feature_table <- subset(feature_table,completeness >= min_feature_completeness)
+  }
+  if(!is.null(min_hypothesis_completeness)){
+    if(type=="complex"){
```

Comment:
Is this really what we want? Why do we keep the bad features if it has one good one? At least for the protein features I think it makes more sense to filter features rather than proteins.